# Практическая работа №5. Запуск метадинамики

Запишу в этот файл команды, которыми запускали метадинамику:

In [2]:
!gmx pdb2gmx -f wildtype_1.pdb -o mutant.gro -p mutant.top -ff amber99sb -water tip3p
!gmx editconf -f mutant.gro -o mutant_box.gro -c -d 1.5 -bt cubic
!gmx grompp -f em.mdp -c mutant_box.gro -p mutant.top -o em.tpr -maxwarn 3
!gmx mdrun -deffnm em -v


               :-) GROMACS - gmx pdb2gmx, 2022.6-plumed_2.9.0 (-:

Executable:   /usr/bin/gmx
Data prefix:  /usr
Working dir:  /home/student/kondratyuk/wildtype
Command line:
  gmx pdb2gmx -f wildtype_1.pdb -o mutant.gro -p mutant.top -ff amber99sb -water tip3p

Using the Amber99sb force field in directory amber99sb.ff

going to rename amber99sb.ff/aminoacids.r2b
Opening force field file /usr/share/gromacs/top/amber99sb.ff/aminoacids.r2b

going to rename amber99sb.ff/dna.r2b
Opening force field file /usr/share/gromacs/top/amber99sb.ff/dna.r2b

going to rename amber99sb.ff/rna.r2b
Opening force field file /usr/share/gromacs/top/amber99sb.ff/rna.r2b
Reading wildtype_1.pdb...
Read '', 146 atoms

Analyzing pdb file
Splitting chemical chains based on TER records or chain id changing.

There are 1 chains and 0 blocks of water and 20 residues with 146 atoms

  chain  #res #atoms

  1 'A'    20    146  

All occupancies are one
All occupancies are one
Opening force field file /usr/share/gromac

In [3]:
!gmx solvate -cp em.gro -cs spc216.gro -o solv.gro -p mutant.top
!gmx grompp -f em.mdp -c solv.gro -p mutant.top -o ions.tpr -maxwarn 3


               :-) GROMACS - gmx solvate, 2022.6-plumed_2.9.0 (-:

Executable:   /usr/bin/gmx
Data prefix:  /usr
Working dir:  /home/student/kondratyuk/wildtype
Command line:
  gmx solvate -cp em.gro -cs spc216.gro -o solv.gro -p mutant.top

Reading solute configuration
Reading solvent configuration

Initialising inter-atomic distances...

         based on residue and atom names, since they could not be
         definitively assigned from the information in your input
         files. These guessed numbers might deviate from the mass
         and radius of the atom type. Please check the output
         files if necessary. Note, that this functionality may
         be removed in a future GROMACS version. Please, consider
         using another file format for your input.

NOTE: From version 5.0 gmx solvate uses the Van der Waals radii
from the source below. This means the results may be different
compared to previous GROMACS versions.

++++ PLEASE READ AND CITE THE FOLLOWING REFERENCE 

In [4]:
!echo '13' | gmx genion -s ions.tpr -o solv_ions.gro -p mutant.top -pname NA -nname CL -neutral -conc 0.15

               :-) GROMACS - gmx genion, 2022.6-plumed_2.9.0 (-:

Executable:   /usr/bin/gmx
Data prefix:  /usr
Working dir:  /home/student/kondratyuk/wildtype
Command line:
  gmx genion -s ions.tpr -o solv_ions.gro -p mutant.top -pname NA -nname CL -neutral -conc 0.15

Reading file ions.tpr, VERSION 2022.6-plumed_2.9.0 (single precision)
Reading file ions.tpr, VERSION 2022.6-plumed_2.9.0 (single precision)
Will try to add 12 NA ions and 14 CL ions.
Select a continuous group of solvent molecules
Group     0 (         System) has 13552 elements
Group     1 (        Protein) has   289 elements
Group     2 (      Protein-H) has   146 elements
Group     3 (        C-alpha) has    20 elements
Group     4 (       Backbone) has    60 elements
Group     5 (      MainChain) has    81 elements
Group     6 (   MainChain+Cb) has    97 elements
Group     7 (    MainChain+H) has    99 elements
Group     8 (      SideChain) has   190 elements
Group     9 (    SideChain-H) has    65 elements
Group    

Fatal error:
The combination of constraints and L-BFGS minimization is not implemented.
Either do not use constraints, or use another minimizer (e.g. steepest
descent).

In [5]:
!gmx grompp -f em_st.mdp -c solv_ions.gro -r solv.gro -p mutant.top -o em_wet.tpr -maxwarn 3
!gmx mdrun -deffnm em_wet -v

               :-) GROMACS - gmx grompp, 2022.6-plumed_2.9.0 (-:

Executable:   /usr/bin/gmx
Data prefix:  /usr
Working dir:  /home/student/kondratyuk/wildtype
Command line:
  gmx grompp -f em_st.mdp -c solv_ions.gro -r solv.gro -p mutant.top -o em_wet.tpr -maxwarn 3

Ignoring obsolete mdp entry 'title'
Ignoring obsolete mdp entry 'cpp'
Ignoring obsolete mdp entry 'ns_type'

NOTE 1 [file em_st.mdp]:
  With Verlet lists the optimal nstlist is >= 10, with GPUs >= 20. Note
  that with the Verlet scheme, nstlist has no effect on the accuracy of
  your simulation.

Setting the LD random seed to -1053061

Generated 2145 of the 2145 non-bonded parameter combinations
Generating 1-4 interactions: fudge = 0.5

Generated 2145 of the 2145 1-4 parameter combinations

Excluding 3 bonded neighbours molecule type 'Protein_chain_A'

Excluding 2 bonded neighbours molecule type 'SOL'

Excluding 1 bonded neighbours molecule type 'NA'

Excluding 1 bonded neighbours molecule type 'CL'

WARNING 1 [file mutan

In [6]:
!gmx grompp -f pr.mdp -c em_wet.gro -r em_wet.gro -p mutant.top -o nvt.tpr -maxwarn 3
!gmx mdrun -deffnm nvt -v


               :-) GROMACS - gmx grompp, 2022.6-plumed_2.9.0 (-:

Executable:   /usr/bin/gmx
Data prefix:  /usr
Working dir:  /home/student/kondratyuk/wildtype
Command line:
  gmx grompp -f pr.mdp -c em_wet.gro -r em_wet.gro -p mutant.top -o nvt.tpr -maxwarn 3

Ignoring obsolete mdp entry 'title'
Ignoring obsolete mdp entry 'cpp'
Ignoring obsolete mdp entry 'ns_type'
Replacing old mdp entry 'nstxtcout' by 'nstxout-compressed'

NOTE 1 [file pr.mdp]:
  nstcomm < nstcalcenergy defeats the purpose of nstcalcenergy, consider
  setting nstcomm equal to nstcalcenergy for less overhead

Setting the LD random seed to -816400393

Generated 2145 of the 2145 non-bonded parameter combinations
Generating 1-4 interactions: fudge = 0.5

Generated 2145 of the 2145 1-4 parameter combinations

Excluding 3 bonded neighbours molecule type 'Protein_chain_A'

turning all bonds into constraints...

Excluding 2 bonded neighbours molecule type 'SOL'

turning all bonds into constraints...

Excluding 1 bonded nei

In [7]:
!gmx grompp -f md.mdp -c nvt.gro -r nvt.gro -p mutant.top -o md.tpr -maxwarn 3

               :-) GROMACS - gmx grompp, 2022.6-plumed_2.9.0 (-:

Executable:   /usr/bin/gmx
Data prefix:  /usr
Working dir:  /home/student/kondratyuk/wildtype
Command line:
  gmx grompp -f md.mdp -c nvt.gro -r nvt.gro -p mutant.top -o md.tpr -maxwarn 3

Ignoring obsolete mdp entry 'title'
Ignoring obsolete mdp entry 'ns_type'

NOTE 1 [file md.mdp]:
  You have set rlist larger than the interaction cut-off, but you also have
  verlet-buffer-tolerance > 0. Will set rlist using verlet-buffer-tolerance.

Setting the LD random seed to -553648385

Generated 2145 of the 2145 non-bonded parameter combinations
Generating 1-4 interactions: fudge = 0.5

Generated 2145 of the 2145 1-4 parameter combinations

Excluding 3 bonded neighbours molecule type 'Protein_chain_A'

turning H bonds into constraints...

Excluding 2 bonded neighbours molecule type 'SOL'

turning H bonds into constraints...

Excluding 1 bonded neighbours molecule type 'NA'

turning H bonds into constraints...

Excluding 1 bonded 

In [8]:
!gmx mdrun -deffnm md -v -nsteps 1000000 -ntomp 4

                :-) GROMACS - gmx mdrun, 2022.6-plumed_2.9.0 (-:

Executable:   /usr/bin/gmx
Data prefix:  /usr
Working dir:  /home/student/kondratyuk/wildtype
Command line:
  gmx mdrun -deffnm md -v -nsteps 1000000 -ntomp 4

Reading file md.tpr, VERSION 2022.6-plumed_2.9.0 (single precision)
Overriding nsteps with value passed on the command line: 1000000 steps, 2e+03 ps
Changing nstlist from 10 to 80, rlist from 0.9 to 1.043


1 GPU selected for this run.
Mapping of GPU IDs to the 2 GPU tasks in the 1 rank on this node:
  PP:0,PME:0
PP tasks will do (non-perturbed) short-ranged interactions on the GPU
PP task will update and constrain coordinates on the CPU
PME tasks will do all aspects on the GPU
Using 1 MPI thread
Using 4 OpenMP threads 


NOTE: The number of threads is not equal to the number of (logical) cpus
      and the -pin option is set to auto: will not pin threads to cpus.
      This can lead to significant performance degradation.
      Consider using -pin on (and -pinoff

In [12]:
!gmx  mdrun -deffnm md -ntomp 4 -v  -plumed plumed.dat

                :-) GROMACS - gmx mdrun, 2022.6-plumed_2.9.0 (-:

Executable:   /usr/bin/gmx
Data prefix:  /usr
Working dir:  /home/student/kondratyuk/wildtype
Command line:
  gmx mdrun -deffnm md -ntomp 4 -v -plumed plumed.dat


Back Off! I just backed up md.log to ./#md.log.3#
Reading file md.tpr, VERSION 2022.6-plumed_2.9.0 (single precision)
Changing nstlist from 10 to 80, rlist from 0.9 to 1.043

1 GPU selected for this run.
Mapping of GPU IDs to the 2 GPU tasks in the 1 rank on this node:
  PP:0,PME:0
PP tasks will do (non-perturbed) short-ranged interactions on the GPU
PP task will update and constrain coordinates on the CPU
PME tasks will do all aspects on the GPU
Using 1 MPI thread
Using 4 OpenMP threads 


NOTE: The number of threads is not equal to the number of (logical) cpus
      and the -pin option is set to auto: will not pin threads to cpus.
      This can lead to significant performance degradation.
      Consider using -pin on (and -pinoffset in case you run multiple

Так поступили с мутантным и wildtype белком. Отличие в plumed.dat - тут для мутанта необходимо было изменить строку ATOMS. Искали С альфа атомы с помощью `grep 'CA' mutant.pdb`

Требовалось отметить изменение энергии:

| Параметр           | Значение          | Атом  |
|---------------------|-------------------|-------|
| **Начальный Fmax**  | 5.307 * 10^3 | 146   |
| **Конечный Fmax**   | 5.225 * 10^2 | 300   |

Требовалось визуально отметить изменеия в системах p_pr.gro и p_si.gro:

| Аспект                | `p_si.gro` (после `gmx genion`)           | `p_pr.gro` (после `gmx grompp`)                      |
|------------------------|-------------------------------------------|------------------------------------------------------|
| **Количество атомов** | `10750` атомов (ослабление воды/иона)       | `10750` атомов (сохранены изменения из `p_si.gro`)    |
| **Координаты**         | Известные позиции воды/ионов (результат `genion`) | Немного изменённые координаты белка/воды (минимизация энергии) |
| **Векторы скоростей**  | **Отсутствуют**                           | **Присутствуют** (добавлены `gmx grompp` с `-f`/.mdp) |
| **Структура белка**     | Атомы с координатами, не с учётом энергии   | Атомы с пониженной энергией (релаксация структуры)     |